src/components/graph/menus/create-node-menu.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import React from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import Menu from '@material-ui/core/Menu';
+import MenuItem from '@material-ui/core/MenuItem';
+
+import ListItemText from '@material-ui/core/ListItemText';
+import Typography from '@material-ui/core/Typography';
+import { useIntl } from 'react-intl';
+import PropTypes from 'prop-types';
+
+const useStyles = makeStyles((theme) => ({
+    menuItem: {
+        padding: '0px',
+        margin: '7px',
+    },
+    listItemText: {
+        fontSize: 12,
+        padding: '0px',
+        margin: '4px',
+    },
+}));
+
+const CreateNodeMenu = ({
+    position,
+    handleClose,
+    handleNodeCreation,
+    handleNodeRemoval,
+    selectedNode,
```

Comment:
The selectedNode prop here is different from the selectedNode of the network-modification-tree. It is the element of the tree on which context menu was triggered, not the curently selectedNode of the tree. So we need to pass it to the menu as its state is not stored in network-modification-tree in order to know on which element the menu functions are to be applied.

Reply:
In gridexplore, we chose to name the element that is target to the contextual menu as 'active', to differentiate it from the 'selected' elements because the is last shorter. If it can help.

Reply:
Good idea, selectedNode prop renamed to activeNode.